src/resonanceReconstruction/breitWigner/multiLevel/Apply/src/call.hpp

```diff
@@ -2,42 +2,61 @@ template< typename Functor >
 decltype(auto)
 operator()( const ENDF::ResonanceRange& range,
             Functor&& callback ) const {
+
   try {
 
-    EnergyRange energyRange{ range.EL() * electronVolts,
-                            range.EH() * electronVolts };
-    auto mlbw = std::get< ENDF::resolved::MLBW >( range.parameters() );
+    const EnergyRange energyRange {
+      range.EL() * electronVolts,
+      range.EH() * electronVolts
+    };
+    const auto mlbw = std::get< ENDF::resolved::MLBW >( range.parameters() );
 
-    if( range.NRO() ){
-      switch( range.NAPS() ){
-      case 0:
-        return callback( build( energyRange, mlbw,
-                                channelRadius( mlbw.lValues().front().AWRI() ),
-                                radius( range.scatteringRadius().value() ) ) );
-      case 1:
-        return callback( build( energyRange, mlbw,
-                                radius( range.scatteringRadius().value() ) ) );
-      case 2:
-        return callback( build( energyRange, mlbw, radius( mlbw.AP() ),
-                                radius( range.scatteringRadius().value() ) ) );
+    if (range.NRO()) {
+      switch (range.NAPS()) {
+        case 0: // fall-through...
+        case 2:
+          return
+            callback( build(
+              energyRange, mlbw,
+              range.NAPS() == 0
```

Comment:
So, I'm not sure why this should work—unless we are not testing all the different switch options, and therefore not catching this. We are `switch`-ing on `range.NAPS()`. Now, in `case 2` (which means `ranges.NAPS() == 2`) we are testing `range.NAPS() == 0`, which is never the case so we always return `radius( mlbw.ap())`

Am I missing something?

Reply:
Yes. :-) Look two lines above the return. So, line 16. range.NAPS() can also be 0 here. And, as someone should with things like this, I commented on the (unusual but allowed) case fall-through (sans break or return).